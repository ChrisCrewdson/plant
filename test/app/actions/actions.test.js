// const _ = require('lodash');
const actions = require('../../../app/actions');

describe('/app/actions', () => {
  test('should create a logout failure', (done) => {
    const actual = actions.logoutFailure();
    expect(actual).toMatchSnapshot();
    done();
  });

  test('should create a logout request', (done) => {
    const actual = actions.logoutRequest();
    expect(actual).toMatchSnapshot();
    done();
  });

  test('should create a logout success', (done) => {
    const actual = actions.logoutSuccess();
    expect(actual).toMatchSnapshot();
    done();
  });

  test('should ensure that simplified actions has all of previous exports', () => {
    const isUpper = letter => letter >= 'A' && letter <= 'Z';
    const originalExports = [
      'CHANGE_ACTIVE_LOCATION_ID',
      'CREATE_PLANT_FAILURE',
      'CREATE_PLANT_REQUEST',
      'CREATE_PLANT_SUCCESS',
      'DELETE_LOCATION_MEMBER',
      'DELETE_LOCATION_WEATHER',
      'DELETE_NOTE_FAILURE',
      'DELETE_NOTE_REQUEST',
      'DELETE_NOTE_SUCCESS',
      'DELETE_PLANT_FAILURE',
      'DELETE_PLANT_REQUEST',
      'DELETE_PLANT_SUCCESS',
      'EDIT_NOTE_CHANGE',
      'EDIT_NOTE_CLOSE',
      'EDIT_NOTE_OPEN',
      'EDIT_PLANT_CHANGE',
      'EDIT_PLANT_CLOSE',
      'EDIT_PLANT_OPEN',
      'LOAD_LOCATIONS_FAILURE',
      'LOAD_LOCATIONS_REQUEST',
      'LOAD_LOCATIONS_SUCCESS',
      'LOAD_NOTES_FAILURE',
      'LOAD_NOTES_REQUEST',
      'LOAD_NOTES_SUCCESS',
      'LOAD_PLANT_FAILURE',
      'LOAD_PLANT_REQUEST',
      'LOAD_PLANT_SUCCESS',
      'LOAD_PLANTS_FAILURE',
      'LOAD_PLANTS_REQUEST',
      'LOAD_PLANTS_SUCCESS',
      'LOAD_UNLOADED_PLANTS_FAILURE',
      'LOAD_UNLOADED_PLANTS_REQUEST',
      'LOAD_UNLOADED_PLANTS_SUCCESS',
      'LOAD_USER_FAILURE',
      'LOAD_USER_REQUEST',
      'LOAD_USER_SUCCESS',
      'LOAD_USERS_FAILURE',
      'LOAD_USERS_REQUEST',
      'LOAD_USERS_SUCCESS',
      'LOGOUT_FAILURE',
      'LOGOUT_REQUEST',
      'LOGOUT_SUCCESS',
      'MODIFY_LOCATION_FAILURE',
      'MODIFY_LOCATION_REQUEST',
      'MODIFY_LOCATION_SUCCESS',
      'UPDATE_PLANT_FAILURE',
      'UPDATE_PLANT_REQUEST',
      'UPDATE_PLANT_SUCCESS',
      'UPSERT_LOCATION_MEMBER',
      'UPSERT_LOCATION_WEATHER',
      'UPSERT_NOTE_FAILURE',
      'UPSERT_NOTE_REQUEST',
      'UPSERT_NOTE_SUCCESS',
      'changeActiveLocationId',
      'createPlantFailure',
      'createPlantRequest',
      'createPlantSuccess',
      'deleteNoteFailure',
      'deleteNoteRequest',
      'deleteNoteSuccess',
      'deletePlantFailure',
      'deletePlantRequest',
      'deletePlantSuccess',
      'editNoteChange',
      'editNoteClose',
      'editNoteOpen',
      'editPlantChange',
      'editPlantClose',
      'editPlantOpen',
      'loadLocationsFailure',
      'loadLocationsRequest',
      'loadLocationsSuccess',
      'loadNotesFailure',
      'loadNotesRequest',
      'loadNotesSuccess',
      'loadPlantFailure',
      'loadPlantRequest',
      'loadPlantsFailure',
      'loadPlantsRequest',
      'loadPlantsSuccess',
      'loadPlantSuccess',
      'loadUnloadedPlantsFailure',
      'loadUnloadedPlantsRequest',
      'loadUnloadedPlantsSuccess',
      'loadUserFailure',
      'loadUserRequest',
      'loadUsersFailure',
      'loadUsersRequest',
      'loadUsersSuccess',
      'loadUserSuccess',
      'logoutFailure',
      'logoutRequest',
      'logoutSuccess',
      'modifyLocationFailure',
      'modifyLocationRequest',
      'modifyLocationSuccess',
      'updatePlantFailure',
      'updatePlantRequest',
      'updatePlantSuccess',
      'upsertNoteFailure',
      'upsertNoteRequest',
      'upsertNoteSuccess',
    ];
    originalExports.forEach((exp) => {
      const [firstLetter] = exp;
      if (isUpper(firstLetter)) {
        expect(typeof actions[exp]).toBe('string');
      } else {
        expect(typeof actions[exp]).toBe('function');
      }
    });
  });
});
