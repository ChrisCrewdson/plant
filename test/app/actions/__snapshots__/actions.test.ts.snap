// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/app/actions should create a logout failure 1`] = `
Object {
  "payload": undefined,
  "type": "LOGOUT_FAILURE",
}
`;

exports[`/app/actions should create a logout request 1`] = `
Object {
  "payload": undefined,
  "type": "LOGOUT_REQUEST",
}
`;

exports[`/app/actions should create a logout success 1`] = `
Object {
  "payload": undefined,
  "type": "LOGOUT_SUCCESS",
}
`;
